### ISSUE TITLE
I always get NULL hash with web3j ethSendRawTransaction,Both send ETH and ERC20 tokens

### ISSUE THREAD

**Issue by xiangyw** [web3j/web3j#1210]
# _Issue_title_
I always get NULL hash with web3j ethSendRawTransaction;
Both send ETH and ERC20 tokens

## _Issue_description_
I am try to send ETH and ERC20 tokens with web3j through mainnet.infura.io

## _Issue_context_
1.the attachment is my code, see method 'transferETHAutoNonce' and 'transferERC20Token'
2.the web3j version is 

        <dependency>
            <groupId>org.web3j</groupId>
            <artifactId>core</artifactId>
            <version>4.5.18</version>
        </dependency>

**Comment by xiangyw** [comment]
The most confusing part is that I can't get any error messages

**Comment by AlexandrouR** [comment]
@xiangyw If you check your account in a blockchain explorer is the transaction created?

**Comment by xiangyw** [comment]
no transaction created and no error message

**Comment by xiangyw** [comment]
i am sure the code is right and gas price be set properly,gas limit is 500000,and nonce is 'transactionCount.getTransactionCount()'
private BigDecimal getDynamicGasPrice(){
        String response = remoteRestTemplate.getForObject("https://ethgasstation.info/json/ethgasAPI.json", String.class);
        JSONObject jsonObject= JSONObject.parseObject(response);
        BigDecimal fastGWEI=new BigDecimal(String.valueOf(jsonObject.get("fast")));
        fastGWEI=fastGWEI.divide(new BigDecimal(10));
        return fastGWEI;
    }

**Comment by xiangyw** [comment]
![image](https://user-images.githubusercontent.com/9930591/82395859-01350f00-9a7f-11ea-8818-fd57afcb1656.png)
I just got null hash, no error message

**Comment by AlexandrouR** [comment]
ok I will try to replicate this.

**Comment by xiangyw** [comment]
checking one of the sender address-->  0x6843f22143f229d00d9ef09f5d4ec14139bed21c
![image](https://user-images.githubusercontent.com/9930591/82395956-4b1df500-9a7f-11ea-914a-db4805688564.png)
no tx created ,the 2 display OUT tx was sent by metamask

**Comment by xiangyw** [comment]
there are other two accounts:
0xc63ce73b7c0114738200676b3ff570fd30b9c92e
0xcdc3df46e200cabc1c0537df5c4ed71efefcb132

**Comment by AlexandrouR** [comment]
hey I think I have solved it. I tried to replicate the issue as much as I could but your code is quite complex. My simple eth transactions were failing as well but I didn't get any error just a null transaction hash. The transaction I created but it fails and is discarded by the node in my case it was a gas issue. I would recommend debugging your code and log this line after the transaction is sent.   
` ethSendTransaction.getError().getMessage());` 
This should give you a sense of what is going wrong.
